core/src/main/java/jenkins/model/ProjectNamingStrategy.java

```diff
@@ -99,6 +101,11 @@ public DefaultProjectNamingStrategy() {
         @Override
         public void checkName(String origName) throws Failure {
             // default - should just do nothing (this is how Jenkins worked before introducing this ExtensionPoint)
+            if (StringUtils.isNotBlank(DEFAULT_PATTERN) && StringUtils.isNotBlank(origName)) {
```

Comment:
I think I am going just to do:

```
@Override
        public void checkName(String origName) throws Failure {
            // default - should just do nothing (this is how Jenkins worked before introducing this ExtensionPoint)
            if (StringUtils.isNotBlank(origName) && !Pattern.matches(DEFAULT_PATTERN, origName)) {
                throw new Failure(Messages.Hudson_JobSinglesQuotesAreProhibited());
            }
        }
```

> why not just check string contains?

I am following the way the pattern is used on the rest of the methods. I think this should be fine.